nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_ClientService.java

```diff
@@ -535,4 +539,16 @@ public Configuration getConfiguration() {
     public byte[] toBytesBinary(String s) {
         return Bytes.toBytesBinary(s);
     }
+
+    @Override
+    public String toTransitUri(String tableName, String rowKey) {
+        if (connection == null) {
+            logger.warn("Connection has not been established, could not create a transit URI. Returning null.");
+            return null;
+        }
+        final Configuration configuration = connection.getConfiguration();
+        // Remove white spaces.
+        final String zkQuorum = configuration.get(HBASE_CONF_ZK_QUORUM).replaceAll("\\s", "");
+        return "hbase://" + zkQuorum + "/" + tableName + (rowKey != null && !rowKey.isEmpty() ? "/" + rowKey : "");
```

Comment:
does it make sense to add znode? Two HBase clusters could use the same ZK, right?
(not sure what would be the best way to represent it though, using ``/znode/table`` could be confusing, no?)

Reply:
I don't think it does. I am not a HBase expert by any means, but I know that you can manually specify the IP address to the HBase master when configuring a Java client. This URL could also be problematic for people attempting a Thrift connection via Python or another client that side steps ZooKeeper.

Reply:
Try this for the master IP:

`connection.getAdmin().getClusterStatus().getMaster()`

That returns a `ServerName` instance which looks like it has everything you need.

[https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/ServerName.html](ServerName javadoc)